Zend/zend_inheritance.c

```diff
@@ -1967,7 +1979,7 @@ static void zend_add_trait_method(zend_class_entry *ce, zend_string *name, zend_
 			 */
 			do_inheritance_check_on_method(
 				existing_fn, fixup_trait_scope(existing_fn, ce), fn, fixup_trait_scope(fn, ce),
-				ce, NULL, /* check_visibility */ 0);
+				ce, NULL, ZEND_INHERITANCE_CHECK_PROTO | ZEND_INHERITANCE_RESET_CHILD_OVERRIDE);
```

Comment:
Is this code reachable when `ZEND_ACC_OVERRIDE` is still set on a parent class method? In that case, we'd try to remove the flag on the parent method. I believe it's not, because linking of the parent class will have already thrown, given `override` was not removed.

Reply:
> Is this code reachable when `ZEND_ACC_OVERRIDE` is still set on a parent class method?

yes. Without ZEND_INHERITANCE_RESET_CHILD_OVERRIDE ``Zend/tests/attributes/override/016.phpt`` is failed.
But may be it should fail. Sometimes our tests prove the weird implementation details or even bugs :)

What do you  think?

Reply:
I also didn't expect the need for ZEND_INHERITANCE_RESET_CHILD_OVERRIDE here...

Reply:
> But may be it should fail. Sometimes our tests prove the weird implementation details or even bugs :)

I can't comment on the implementation, due to my lack of knowledge about the engine. But I can comment as the RFC author:

The behavior of `Zend/tests/attributes/override/016.phpt` is correct: If the method is required by an abstract trait method, then `#[\Override]` is satisfied. `#[\Override]` basically acts as a counterpart to the LSP checks. If the LSP check would fail when the signature changes, then `#[\Override]` is satisfied.

The case that is tested by 016.phpt is also mentioned in the RFC:

> The attribute is ignored on traits, but:
> - Abstract methods in a used trait satisfy #[\Override].

Reply:
I think this code is correct. I was only commenting on the possibility of this code being reachable if the checked method is an inherited method. However, I don't believe that's possible, because then `override` would not have been removed from the parent, which should have already made it fail. Hence, the method here should always come from the current class (if `override` is still set).

Reply:
There is no ``parent`` in that test.

Reply:
@dstogov I tried to make it fail with parent but was unable to, because of what I mentioned above:

> because then override would not have been removed from the parent, which should have already made it fail.

The parent class cannot successfully link with `override` staying attached on the method.